src/components/common/UserProfileEdit/UserAvatarUploader.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState } from 'react';
+import { defineMessages } from 'react-intl';
+
+import { FileReaderFile } from '~shared/FileUpload';
+import AvatarUploader from '~shared/AvatarUploader';
+import UserAvatar from '~shared/UserAvatar';
+import { InputStatus } from '~shared/Fields';
+
+import { User } from '~types';
+import { useUpdateUserMutation } from '~gql';
+import { useAppContext } from '~hooks';
+
+import styles from './UserAvatarUploader.css';
+
+const displayName = 'common.UserProfileEdit.UserAvatarUploader';
+
+const MSG = defineMessages({
+  uploaderLabel: {
+    id: `${displayName}.uploaderLabel`,
+    defaultMessage: 'At least 250x250px, up to 1MB, .png or .svg',
+  },
+  avatarFileError: {
+    id: `${displayName}.avatarFileError`,
+    defaultMessage: 'This filetype is not allowed or file is too big',
+  },
+});
+
+interface Props {
+  /** Current user */
+  user: User;
+}
+
+const UserAvatarUploader = ({
+  user,
+  user: { walletAddress, profile },
+}: Props) => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { __typename, ...userProfile } = profile || {};
```

Comment:
hmm.... why  was there a need to remove `__typename` here ? It doesn't really make sense to me

Reply:
it gives an error otherwise
<img width="636" alt="Screenshot 2022-12-07 at 16 46 20" src="https://user-images.githubusercontent.com/34057551/206144984-0cdad0d6-984d-4e47-bae5-0c14302e91d0.png">

Reply:
Apollo will, by default, add `__typename` to queries as it needs it for caching and normalisation. And we will need to remove it, presumably not just here, but wherever we want to use the object from the response in a mutation. So I suggest we create an util function to do it for us so that we don't have to destructure and disable eslint every time.

Reply:
great idea @jakubcolony - implemented